### PR TITLE
Break instance init to two steps to access wasm global on start function error

### PR DESCRIPTION
**Pull request by ailisp**
break instance init to create instance step and call start function step. This is because if start function went into error, we don't have an instance in near, therefore cannot access the gas counter global in near.

### CODE REVIEW

File: lib/runtime-core/src/instance.rs [hunk 1]
```diff
@@ -118,32 +118,32 @@ impl Instance {
...
+                            },
+                            runnable_module: module.runnable_module.clone(),
+                        });
+                        Some(())
+                    } else {
+                        None
+                    }
+                };
+                inner.code_version_pushed = push_code_version_logic().is_some();
+            }
```
Comment (matklad) [inline lib/runtime-core/src/instance.rs:146]: Let's remove unrelated ws changes.

**Review by matklad — COMMENTED** [review]
LGTM!

Let's keep the PR open until we are confident that we are merging #4688 (as in, there might be some unknowns there which would require *more* changes to wasmer).

**Comment by nagisa** [comment]
I believe this is no longer a concern that needs to be fixed, correct? As we pass in a structure for fast gas counting when creating an instance and running the start function will update counters therein?

**Comment by matklad** [comment]
Yeah, I believe this is correct!
